fs_mark/fs_mark.py

```diff
@@ -0,0 +1,214 @@
+import trunner
+import time
+import os
+import yaml
+import re
+import pexpect
+import timing_data as t_data
+import psh.tools.psh as psh
+
+from datetime import datetime
+from trunner.ctx import TestContext
+from trunner.dut import Dut
+from trunner.types import TestResult, TestSubResult, Status
+
+# Targets with dummyfs
+dummy_list = ['armv7m7-imxrt105x-evk', 'armv7m7-imxrt106x-evk',
+              'armv7m7-imxrt117x-evk', 'armv7m4-stm32l4x6-nucleo']
+
+
+def clean(dut, test_name, ctx: TestContext) -> int:
+    global dirs
+
+    if ctx.target.name in dummy_list:
+        return 0
+
+    if '-k' in test_name or '-L' in test_name or '-F' in test_name:
+        args = test_name.split()
+        args = args[1:]
+
+        def get_dir(x):
+            if args[x[0] - 1] == '-d':
+                return x[1]
+
+        dirs = list(map(lambda x: get_dir(x), enumerate(args)))
+        dirs = [x for x in dirs if x is not None]
+
+        dut.expect(re.escape(ctx.target.shell_prompt), timeout=10)
+        psh._send(dut, f'/bin/fs_mark_clean {" ".join(dirs)}')
+        try:
+            idx = dut.expect([re.escape(ctx.target.shell_prompt),
+                              r".+?Error in remove_dir_recursive.+?"], timeout=1800)
+
+            if idx == 1:
+                return -1
+            elif idx == 0:
+                return 0
+        except pexpect.TIMEOUT:
+            return -1
+    else:
+        return 0
+
+
+def harness(dut: Dut, ctx: TestContext, result: TestResult):
+    target = trunner.ctx.target.name
+    logs_path = '~/remote/logs/fs_mark_logs_test_old'
+    test_status = Status.OK
+
+    subresult = None
+    test_name = None
+    got_time = False
+    first_loop = True
+
+    f_use = None
+    count = None
+    size = None
+    files_sec = None
+    app_overhead = None
+
+    start_time = None
+    end_time = None
+    loop_time = 1200
+
+    NAME = r".+?#  (?P<name>(/bin/)?fs_mark.+?)\r+\n"
+    MSG_LINE = r"(?P<line>(\s+\d+){3}.+?)\r+\n"
+    NO_SPC = r"Insufficient free space.+?\r+\n"
+    END = r"Average Files/sec:.+?\r+\n"
+    EIO_ERR = r".+?EIO.+?"
+    ENOSPC_ERR = r".+?ENOSPC.+?"
+    ENOMEM_ERR = r".+?ENOMEM.+?"
```

Comment:
Maybe it would be better to do it in one regex, and catch all these cases, and then use it in one case, in the message.
For example, what I mean from another existing harrnes:
![image](https://github.com/phoenix-rtos/phoenix-rtos-tests/assets/116297248/7ecede1b-27e3-4712-8872-d518db87f1e8)